pkg/build/pipeline.go

```diff
@@ -182,6 +182,7 @@ func buildEvalRunCommand(pipeline *config.Pipeline, debugOption rune, workdir st
 [ -d '%s' ] || mkdir -p '%s'
 cd '%s'
 %s
+[ -n "$(jobs -p)" ] && kill $(jobs -p)
```

Comment:
negative logic is harder to read, but with shell it is the rigth thing to use.

if for some reason, 'jobs -p' fails and doesn't output anything, then you dont kill any jobs. 
i can agree that might sound unlikely.  however, doing it this way means you swallow errors kill.

heres an example:

```
$ sh -ec '
> sleep 100s &
> [ -n "$(jobs -p)" ] && kill adf $(jobs -p)
> echo still here, gonna exit 0 now.
> exit 0
> '
still here, gonna exit 0 now.
```

Reply:
Better is to do something like this:

```
set -e
jobs=$(jobs -p)
if [ -n "$jobs" ]; then
  echo "killing jobs $jobs"
  kill -9 $jobs
fi
exit 0
```

this way `set -e` will catch the failures at each step.

also, re-enable `set -e` becuse the executed code block could have disabled it.

huh... i guess that code block could also have exited zero, which would mean this code would never execute. that needs fixing as well.  `trap` would be an option, but i know some of the tests and things use trap themselves. (sorry, probably most are from me).

Reply:
also, the location of this code (in "generic" vs in ssh) will affect other runners.

Reply:
I'm changing to the if/else like you suggested 
this should not be a problem with other runners, as this is what happens anyway with the one-shot execution

Reply:
also moved to trap, this trap is executed both first and last so ensure this cleanup is triggered